src/api/spec/controllers/webui/comments_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe Webui::CommentsController do
+  let(:user) { create(:confirmed_user, login: 'luck') }
+  let(:user_tom) { create(:confirmed_user, login: 'tom') }
+  let(:admin) { create(:admin_user, login: 'Admin') }
+  let(:project) { create(:project, name: 'openSUSE') }
+  let(:package) { create(:package, name: 'osc', project: project) }
+  let(:bs_request) { create(:bs_request, description: "Please take this", creator: user.login) }
+  let(:comment) { create(:comment_project, user: user) }
+  let(:tom_comment) { create(:comment_project, user: user_tom) }
+
+  describe 'POST #create' do
+    before do
+      login user
+    end
+
+    context 'saving a comment of a project' do
+      before do
+        post :create, params: { comment: { body: 'This project is AWESOME!', commentable_type: project.class, commentable_id: project.id } }
+      end
+
+      it { expect(flash[:notice]).to eq('Comment was successfully created.') }
+    end
+
+    context 'saving a comment of a package' do
+      before do
+        post :create, params: { comment: { body: 'Wonderful Package', commentable_type: package.class, commentable_id: package.id } }
+      end
+
+      it { expect(flash[:notice]).to eq('Comment was successfully created.') }
+    end
+
+    context 'saving a comment of a request' do
+      before do
+        post :create, params: { comment: { body: 'Awesome Request', commentable_type: bs_request.class, commentable_id: bs_request.id } }
+      end
+
+      it { expect(flash[:notice]).to eq('Comment was successfully created.') }
+    end
+
+    context 'could not save a comment without commentable object' do
+      before do
+        post :create, params: { comment: { body: 'Wonderful Package' } }
+      end
+
+      it { expect(flash[:error]).to match('Comment can\'t be saved') }
```

Comment:
Why is the flash missing:

```
comment.errors.full_messages.to_sentence
```

And furthermore, this looks more like model test (validation). I would only test for the flash and nothing got created (look at the title of the test).